client/app/bundles/course/assessment/pages/AssessmentStatistics/StudentMarksPerQuestionTable.tsx

```diff
@@ -0,0 +1,289 @@
+import { FC, ReactNode } from 'react';
+import { defineMessages } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import { Box, Chip } from '@mui/material';
+import palette from 'theme/palette';
+import { MainSubmissionInfo } from 'types/course/statistics/assessmentStatistics';
+
+import { workflowStates } from 'course/assessment/submission/constants';
+import Link from 'lib/components/core/Link';
+import GhostIcon from 'lib/components/icons/GhostIcon';
+import Table, { ColumnTemplate } from 'lib/components/table';
+import { DEFAULT_TABLE_ROWS_PER_PAGE } from 'lib/constants/sharedConstants';
+import { useAppSelector } from 'lib/hooks/store';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import { getClassNameForMarkCell } from './ColorGradationLevel';
+import { getAssessmentStatistics } from './selectors';
+
+const translations = defineMessages({
+  name: {
+    id: 'course.assessment.statistics.name',
+    defaultMessage: 'Name',
+  },
+  group: {
+    id: 'course.assessment.statistics.group',
+    defaultMessage: 'Group',
+  },
+  totalGrade: {
+    id: 'course.assessment.statistics.totalGrade',
+    defaultMessage: 'Total',
+  },
+  grader: {
+    id: 'course.assessment.statistics.grader',
+    defaultMessage: 'Grader',
+  },
+  searchText: {
+    id: 'course.assessment.statistics.searchText',
+    defaultMessage: 'Search by Group or Grader Name',
```

Comment:
Why not searchable by student name?

Reply:
the first requirement is only to be searchable by group or grader name (basically for the sake of filtering). But after thinking again, there's nothing wrong in having student name being searchable, and hence I added it already